src/Microsoft.Health.Blob/Registration/BlobClientRegistrationExtensions.cs

```diff
@@ -132,15 +132,16 @@ public static IServiceCollection AddBlobServiceClient(this IServiceCollection se
                     if (string.Equals(options.Credential, "managedidentity", StringComparison.OrdinalIgnoreCase))
                     {
                         clientBuilder = builder
-                            .AddBlobServiceClient(new Uri(options.ConnectionString))
-                            .WithCredential(new DefaultAzureCredential(new DefaultAzureCredentialOptions { ManagedIdentityClientId = options.ClientId }));
+                            .AddBlobServiceClient(new Uri(options.ConnectionString));
                     }
                     else
                     {
                         clientBuilder = builder
                             .AddBlobServiceClient(options.ConnectionString);
                     }
 
+                    clientBuilder = clientBuilder.WithCredential(new DefaultAzureCredential(options.Credentials));
```

Comment:
Given your other comment, I think you were right originally to keep this only in the managed identity scenario, as the credentials are included as part of the `string` otherwise.

Reply:
Makes sense! Reverted my last change